xesmf/frontend.py

```diff
@@ -1017,6 +1034,30 @@ def __init__(
             self._lon_out_name = 'lon'
             self._lat_out_name = 'lat'
 
+        # Check length of polys segments, issue warning if too long
```

Comment:
Do you think you could make use of `util.split_polygons_and_holes` to get all sub-polygons ? I think it could reduce the length of this code.